test/unittests/test_dd_synthesis_dc.cpp

```diff
@@ -0,0 +1,68 @@
+#include "algorithms/synthesis/dd_synthesis.hpp"
+#include "algorithms/synthesis/encoding.hpp"
+#include "core/io/pla_parser.hpp"
+#include "dd/Simulation.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace dd::literals;
+using namespace syrec;
+
+class TestDDSynthDc: public testing::TestWithParam<std::string> {
+protected:
+    TruthTable                     tt{};
+    std::string                    testCircuitsDir = "./circuits/";
+    std::unique_ptr<dd::Package<>> dd              = std::make_unique<dd::Package<>>(15U);
+    std::string                    fileName;
+
+    void SetUp() override {
+        fileName = testCircuitsDir + GetParam() + ".pla";
+    }
+};
+
+INSTANTIATE_TEST_SUITE_P(TestDDSynth, TestDDSynthDc,
+                         testing::Values(
+                                 "huff_1",
+                                 "dcX2bit",
+                                 "dc3bit",
+                                 "dc3bitNew",
+                                 "dcX4bit",
+                                 "sym6_32"),
+                         [](const testing::TestParamInfo<TestDDSynthDc::ParamType>& info) {
+                             auto s = info.param;
+                             std::replace( s.begin(), s.end(), '-', '_');
+                             return s; });
+
+TEST_P(TestDDSynthDc, GenericDDSynthesisDcTest) {
+    EXPECT_TRUE(readPla(tt, fileName));
+    extend(tt);
+
+    encodeHuffman(tt);
+
+    const auto ttDD = buildDD(tt, dd);
+    EXPECT_TRUE(ttDD.p != nullptr);
+
+    DDSynthesizer synthesizer(tt.nInputs());
+    const auto&   qc = synthesizer.synthesize(ttDD, dd);
+
+    EXPECT_TRUE(!qc.empty());
+
+    std::vector<dd::vEdge> inEdges;
+    inEdges.reserve(tt.size());
+    std::vector<std::string> outputStrings;
+    outputStrings.reserve(tt.size());
+
+    for (auto const& [input, output]: tt) {
+        inEdges.emplace_back(dd->makeBasisState(static_cast<dd::Qubit>(tt.nInputs()), input.toBoolVec()));
+        outputStrings.emplace_back(output.toString());
+    }
+
+    for (auto in = 0U; in < inEdges.size(); in++) {
+        const auto vOut       = dd::simulate(&qc, inEdges[in], dd, 1);
+        const auto vOutString = vOut.begin()->first;
+        EXPECT_TRUE(TruthTable::Cube::equalDcCube(vOutString, outputStrings[in]));
+    }
```

Comment:
This simulation functionality should probably not reside in a single test. The SyReC project already has some means for simulating circuits in the `algorithms/simulation` directory. I'd argue that the functionality to take a `qc::QuantumComputation` and produce a `TruthTable` should either reside there, e.g., as a function
```c++
[[nodiscard]] auto buildTruthTable(const qc::QuantumComputation& qc, TruthTable& tt) -> void;
```
or should be an additional constructor in the `TruthTable` class. Although, I'd probably prefer the first solution as it separates the simulation from the actual data structure.

Then, this tests should just compare both TTs without considering don't care values.

Reply:
I'd say no in this case. The function takes a quantum circuit and produces a TT. Technically you wouldn't need DDs at all. You could just adapt the `simple_simulation` scheme to work on `qc::QuantumComputation`s. Actually, that might be way more efficient than simulating with DDs in this case.

Reply:
I understand. however, I think there is some work that is supposed to be done. The simple_simulation uses dynmic_bitset and some other stuff which is syrec specific. I'm pretty sure it can adapted later on.

Can we please deal with this in another PR?

Reply:
Adapting the `simpleSimulation` can be deferred to another PR. However, the remaining comments still need to be addressed properly.